src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -216,7 +218,7 @@ private async Task<BundleResponse> ExecuteTransactionForAllRequests(Hl7.Fhir.Mod
             catch (TransactionAbortedException)
             {
                 _logger.LogError("Failed to commit a transaction. Throwing BadRequest as a default exception.");
-                throw new TransactionFailedException(Api.Resources.GeneralTransactionFailedError, HttpStatusCode.BadRequest);
+                throw new FhirTransactionFailedException(Api.Resources.GeneralTransactionFailedError, HttpStatusCode.BadRequest);
```

Comment:
I think it'd be worth while adding a comment to `FhirTransactionFailedException` that notes it relates to a FHIR Bundle Transaction